clickplc/driver.py

```diff
@@ -35,14 +37,25 @@ class ClickPLC(AsyncioModbusClient):
         'ds': 'int16',   # (D)ata register (s)ingle
         'dd': 'int32',   # (D)ata register, (d)ouble
         'dh': 'int16',   # (D)ata register, (h)ex
-        'df': 'float',   # (D)ata register (f)loating point
+        'df': 'float',   # (D)ata register (f)loating
+        'xd': 'int32',   # Input Register
+        'yd': 'int32',   # Output Register
         'td': 'int16',   # (T)imer register
         'ctd': 'int32',  # (C)oun(t)er Current values, (d)ouble
         'sd': 'int16',   # (S)ystem (D)ata register, single
         'txt': 'str',    # ASCII Text
     }
 
-    def __init__(self, address, tag_filepath='', timeout=1):
+    def __init__(self,
+                 address,
+                 tag_filepath='',
+                 timeout=1,
+                 interfacetype: Literal["TCP", "Serial"] = 'TCP',
```

Comment:
Your approach would likely work fine  (I haven't tested it).

However, this isn't the only driver I maintain and I prefer them to have similar patterns.  In this case, it's pretty straightforward to determine if we are using serial or TCP and just select the correct client automatically:
See https://github.com/alexrudd2/alicat/blob/master/alicat/driver.py#L33-L48